lib/notifications/notification.ex

```diff
@@ -131,48 +123,31 @@ defmodule Notifications.Notification do
           Skate.Repo.one(
             from(n in DbNotification,
               join: bm in assoc(n, :bridge_movement),
-              where: bm.inserted_at > ^cutoff_time,
+              preload: [bridge_movement: bm],
+              where:
+                bm.inserted_at > ^cutoff_time and bm.status == ^bridge_movement_values[:status],
               order_by: [desc: bm.inserted_at],
               limit: 1
             )
           )
 
         if existing_record do
-          {:existing_record, existing_record}
+          existing_record
         else
           changeset =
             DbNotification.bridge_movement_changeset(
               %DbNotification{created_at: created_at},
               bridge_movement_values
             )
 
-          {:ok, new_record} = Skate.Repo.insert(changeset)
+          {:ok, new_record} = Skate.Repo.insert(changeset, returning: true)
           link_notification_to_users(new_record.id, @chelsea_st_bridge_route_ids)
-          {:new_record, new_record}
+          log_creation(new_record)
+          new_record
         end
       end)
 
-    {reason, end_time} =
-      if bridge_movement_values.status == :raised do
-        {:chelsea_st_bridge_raised, bridge_movement_values.lowering_time}
-      else
-        {:chelsea_st_bridge_lowered, nil}
-      end
-
-    notification = %__MODULE__{
-      id: db_record.id,
-      created_at: db_record.created_at,
-      reason: reason,
-      route_ids: @chelsea_st_bridge_route_ids,
-      run_ids: [],
-      trip_ids: [],
-      start_time: db_record.created_at,
-      end_time: end_time,
-      state: :unread
-    }
-
-    if source == :new_record, do: log_creation(notification)
```

Comment:
**Praise**: It took me a little while to wrap my mind around why we needed the whole `source` thing. Removing that (and moving the log line to the right spot) makes this a lot easier to understand!